src/plugins/mf_priority.cpp

```diff
@@ -824,6 +863,59 @@ static int update_queue_cb (flux_plugin_t *p,
 }
 
 
+/*
+ *  Check for an updated bank and validate it for a user/bank; if the
+ *  user/bank does not have access to the bank they are trying to update
+ *  their job for, reject the update and keep the job under its current bank.
+ *
+ *  Also check the active jobs and running jobs limits for the new bank; if the
+ *  new bank is currently at its max active jobs or max running jobs limit,
+ *  reject the update and keep the job under its current bank.
+ */
+static int update_bank_cb (flux_plugin_t *p,
+                           const char *topic,
+                           flux_plugin_arg_t *args,
+                           void *data)
+{
+    int userid;
+    char *bank = NULL;
+    Association *a;
+
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:i, s:s}",
+                                "userid", &userid,
+                                "value", &bank) < 0) {
+        return flux_jobtap_error (p, args, "unable to unpack bank arg");
+    }
+
+    // look up association
+    a = get_association (userid, bank, users, users_def_bank);
+    if (a == nullptr)
+        return flux_jobtap_reject_job (p,
+                                       args,
+                                       "cannot find user/bank or "
+                                       "user/default bank entry "
+                                       "for uid: %i", userid);
+
+    if (a->max_active_jobs > 0 && a->cur_active_jobs >= a->max_active_jobs)
+        // new bank is already at its max active jobs limit; reject update
+        return flux_jobtap_error (p,
+                                  args,
+                                  "mf_priority: new bank is already at "
+                                  "max-active-jobs limit");
+
+    if (a->max_run_jobs > 0 && a->cur_run_jobs == a->max_run_jobs)
+        // new bank is already at its max running jobs limit; reject update
```

Comment:
It may be helpful to expand this comment. If a user can submit a job to a bank that is already it is maximum running jobs limit, then it might be confusing as to why they can't also update a job to that bank. I'm guessing this has something to do with the fact that jobs over the max-run-jobs limit are held in DEPEND state, and there isn't a way to kick a job back to that state?

A comment here detailing that reason (or the real reason if I got it wrong), and a more specific error returned to the user, e.g. "updating to bank %s at max-run-jobs limit not allowed. Try again later" or similar.

Reply:
That's a great suggestion, thanks! I went ahead and edited this error message to be a little more descriptive as well as added a slightly more detailed comment.